sklearn/ensemble/_hist_gradient_boosting/gradient_boosting.py

```diff
@@ -318,13 +375,66 @@ def fit(self, X, y):
         del self._in_fit  # hard delete so we're sure it can't be used anymore
         return self
 
+    def _has_been_fitted(self):
+        return len(getattr(self, '_predictors', [])) > 0
+
+    def _clear_state(self):
+        """Clear the state of the gradient boosting model."""
+        if hasattr(self, 'train_score_'):
+            del self.train_score_
+        if hasattr(self, 'validation_score_'):
+            del self.validation_score_
+        if hasattr(self, '_rng'):
+            del self._rng
+
+    def _get_small_trainset(self, X_binned_train, y_train,
+                            seed, subsample_size=10000):
+        """Compute the indices of the subsample set and return this set.
+
+        For efficiency, we need to subsample the training set to compute scores
+        with scorers. Also note that the returned indices are not expected to
+        be the same between different calls to fit() in a warm start context,
+        since the rng may have been consumed.
+
+        Parameters
+        ----------
```

Comment:
I rarely say that but in this case I think we can get rid of the Parameters and Returns parts.